docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,302 @@
+we are also not interested in isolating containers in a pod from one another, the SELinux context
+should be shared by all containers in a pod to facilitate isolation from the containers in other
+pods and sharing resources amongst all the containers of a pod.
+
+### Pod-level supplemental groups
+
+The [generalized non-root and SELinux support for volumes](https://github.com/kubernetes/kubernetes/pull/12944)
+pull request describes in detail the cases for sharing volumes using supplemental groups.  The
+easiest way to share resources among containers running as different UID/primary GID combinations is
+to make them owned by a supplemental group that all containers run as.  Therefore, there should be a
+pod-level supplemental group field that cannot be overridden in containers.
+
+### Backward compatibility
+
+The existing V1 API should be backward compatible with the new API.  Backward compatibility is
+defined as:
+
+1.  Any API call (e.g. a structure POSTed to a REST endpoint) that worked before your change must
+    work the same after your change.
+2.  Any API call that uses your change must not cause problems (e.g. crash or degrade behavior) when
+    issued against servers that do not include your change.
+3.  It must be possible to round-trip your change (convert to different API versions and back) with
+    no loss of information.
+
+This essentially means that we must keep all the existing fields in place in the V1 API and make
+only additive changes.  So, in order to design the correct API changes, we must:
+
+1.  Retain the current container level `SecurityContext`
+2.  Define which fields must be controllable at the container level; this becomes the new 
+    `ContainerSecurityContext`
+3.  Define which fields must be controllable at the pod level; this becomes the new
+    `PodSecurityContext`; fields in the `PodSecurityContext` are one of the following:
+    1.  New fields (SupplementalGroupId)
+    2.  Fields that are currently part of the `PodSpec`
+    3.  Fields that are currently part of the container level `SecurityContext` and are not part
+        of the `ContainerSecurityContext`
+    4.  Fields that are currently part of the container level `SecurityContext` and **are** in the
+        `ContainerSecurityContext` -- these will be fields of the `ContainerDefaults`
+
+Let's break down the fields of the `PodSecurityContext` and examine the backward compatibility rules
+for each class of field:
+
+1.  Fields that are purely additive have no backward compatibility concerns
+2.  For fields that are moving from the `PodSpec` to the `PodSecurityContext`:
+    1.  If the new field is set and the deprecated field is unset, the deprecated field is defaulted
+        to the value of the new field
+    2.  If the deprecated field is set and the new field is unset, the new field is defaulted to the
+        value of the deprecated field
+    3.  If the deprecated field and new field are both set, their values must match
```

Comment:
Depends on where you are.  On the way in, the deprecated field takes priority ALWAYS.  Regardless of whether they match.  Otherwise, the client might get an object from the server, mutate the field they know about (but ignore the field they don't know about) and send it back to the server.  The server sees the old value of the new field and the new value of the deprecated field.  

You have to define the rules as "old wins if old is specified at all" when converting a serialized value into the external representation.